engine/src/cmd/unit_csv.cpp

```diff
@@ -120,19 +108,21 @@ void AddMeshes(std::vector<Mesh *> &xmeshes,
         Flightgroup *fg,
         vector<unsigned int> *counts) {
     string::size_type where, when, wheresf, wherest, ofs = 0;
+
+    // Clear counts vector
     if (counts) {
         counts->clear();
     }
+
+    // Reserve space in counts and xmeshes vectors based on number of opening braces
     {
-        int nelem = 0;
-        while ((ofs = meshes.find('{', ofs)) != string::npos) {
-            nelem++, ofs++;
-        }
+        // This code is proven to work for one mesh.
```

Comment:
@royfalk can you expand on this comment a little? Has the code not been tested with more than one mesh?

Reply:
If you look at the json, you'll see a mesh field like this:
"Mesh": "{turretlauncher.bfxm;;}"
Originally, they planned for multi-mesh support like this:
"Mesh": "{turretlauncher.bfxm;;}{another_mesh.bfxm;;}"

AFAIK, there isn't an example of multi-mesh for me to test, hence the comment. As for the old code, the original devs were a lot looser with the quality of what they put in. Specifically, they had a tendency to put half baked projects in. This makes it difficult to guess what some code is used for.